feat(protocol): add note about partial updates (#398)
NordicSemiconductor/asset-tracker-cloud-docs: docs/cloud-protocol/DeviceDataAndConfiguration.rst

   :local:
   :depth: 2

> The schemas referenced below define the data format used by the device to encode data.

Reviewer: ```suggestion
The schemas referenced on this page define the data format used by the device to encode data.
```

> Currently :ref:`JSON is the only supported encoding format <json-format>`.

Reviewer: ```suggestion
Currently, :ref:`JSON <json-format>` is the only supported encoding format.
```
Just a proposal. Works as it is, but I'd rather have it like this.

While the schemas mark properties as required, the device can always send partial updates.

> For example, the device might initially after boot report the full roaming information, but later only send the changed values (typically RSRP) to the cloud in order conserve data and power.

Reviewer: ```suggestion
For example, the device might initially report the full roaming information after boot, but later only send the changed values (typically RSRP) to the cloud to conserve data and power.
```
Should we open the abbreviation?

> For example, the device might initially report the full roaming information after boot, but later only send the changed values (typically RSRP) to the cloud to conserve data and power.

Reviewer: What about the abbreviation RSRP? Would it be worth opening?

Author: It's explained in the schema files, but good idea to explain it here.

> For example, the device might initially report the full roaming information after boot, but later only send the changed values (typically RSRP) to the cloud to conserve data and power.

Author: ```suggestion
For example, the device might initially report the full roaming information after boot, but later only send the changed values, for example Reference Signal Received Power (RSRP), to the cloud to conserve data and power.
```

> It is the responsibility of the cloud to store the updates in a way that full objects can be constructed from partial updates, when needed.

Reviewer: Do we need to say "when needed"?

Author: Not really, it's implied.

> It is the responsibility of the cloud to store the updates in a way that full objects can be constructed from partial updates, when needed.

Author: ```suggestion
It is the responsibility of the cloud to store the updates in a way that full objects can be constructed from partial updates.
```
